packages/messaging/ios/RNFBMessaging/RNFBMessaging+AppDelegate.m

```diff
@@ -123,6 +123,18 @@ - (void)application:(UIApplication *)application
     completionHandler(UIBackgroundFetchResultNoData);
     return;
   }
+
+  // If the notification is a probe notification, always call the completion 
```

Comment:
is there some documentation about what a probe notification is? First I have heard of it! It seems to be detectable (via your `data && data[@"warning"]` test but I'd love a canonical source for that test as nothing came up in my attempt at a search

Reply:
Good question & sorry if I put pressure on you :) 
This can wait for you to be back to normal, I just would hate this bug to be in the codebase.

Regarding the probe notification, I haven't found it through the documentation but rather through inspecting the Firebase iOS SDK code. 

https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/SystemService/FIRAuthNotificationManager.m#L105-L110

https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/SystemService/FIRAuthNotificationManager.m#L45-L48

I looked at the blame log and this was part of the iOS Firebase SDK's first commit ever. 

The fix has been found through dissecting the code until I found most minimal cause of the bug, and then got a matching fix.

Perhaps @paulb777 (the original author of the SDK) could give us some details?

Reply:
I did the original open sourcing, but not the authoring so I can't help with the purpose of FirebaseAuth probe notifications, but maybe @renkelvin can?

Reply:
Hi all, I'm not the author either, while my understanding is that this key is to help Firebase Auth to decide whether they should handle the notification or let the App handle it. See https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/SystemService/FIRAuthNotificationManager.m#L150

Reply:
thanks @paulb777 @renkelvin! This definitely brings me back to when I found the fix. 

@mikehardy I should have taken notes but my understanding based on Paul & Chuanr's comments is that we should always callback when its a probe notification since that is what the SDK does when it has pending callbacks:

https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/SystemService/FIRAuthNotificationManager.m#L150-L158

Moreover, there would be no side effect to callback the notification handler since it is effectively empty:
https://github.com/firebase/firebase-ios-sdk/blob/eb30b6fdd809cbfe3a9eb7f3ccf1dd46326bef2d/FirebaseAuth/Sources/SystemService/FIRAuthNotificationManager.m#L116-L117

Therefore the thinking is:

1. we get a probe notification; and RNFB can't handle it.
2. assuming that detox waits for completion handlers to be called back to say things are complete (not an unfair assumption tbh); we can't just drop the notification and never call the completion handler. 
3. Even if we redundantly call back the completion handler --> there is no side effect.

Reply:
@Salakar or @Ehesp just dropping you a notification that links you into some really interesting firebase messaging / auth race condition that is deep enough / subtle enough it probably lives in FlutterFire too - you might take a look. I hate to create new notification pings just for gratitude when we probably get too many, but I'll do it anyway, thanks @paulb777 and @renkelvin for the assist here, much appreciated.